skill-manager/skill_manager/routers/skill.py

```diff
@@ -72,6 +72,40 @@ async def get_skills(request: Request):
     )
     return skills
 
+@router.get(
+    "/dataset/{dataset}",
+    response_model=List[Skill],
+)
+async def get_skills_by_dataset(request: Request, dataset: str = None):
+    """Returns all the skills that were trained on the given dataset."""
+    mongo_query = {"published": True}
+    if has_auth_header(request):
+        payload = await get_payload_from_token(request)
+        user_id = payload["username"]
+        # make a mongo query to get all the skills that the user has access to and are published and trained on the given dataset
+        mongo_query = {
+            "$and": [
+                {"data_sets": dataset},
+                {
+                    "$or": [
+                        mongo_query,
+                        # {"user_id": user_id}, # this user id is not the one that logged in
```

Comment:
but it should be the user id of the user that actually triggered the request, right? Could we make the user_id another parameter of this endpoint? Ie the meta-skill will call this endpoit with the token (that has the meta-skill as user_id), however, the meta-skill can know who called it and pass the user_id to this endpoint too